app/src/common/shared/org/mozilla/vrbrowser/telemetry/GleanMetricsService.java

```diff
@@ -53,6 +63,103 @@ public static void stop() {
         Glean.INSTANCE.setUploadEnabled(false);
     }
 
+    public static void startPageLoadTime() {
+       // TODO: Blocked by Bug 1595914.
+       // pageLoadingTimerId = Pages.INSTANCE.getPageLoad().start();
+    }
+
+    public static void stopPageLoadTimeWithURI(String uri) {
+        // TODO: Blocked by Bug 1595914.
+        // Pages.INSTANCE.getPageLoad().stopAndAccumulate(pageLoadingTimerId);
+
+        try {
+            URI uriLink = URI.create(uri);
+            if (uriLink.getHost() == null) {
+                return;
+            }
+
+            if (domainMap.add(UrlUtils.stripCommonSubdomains(uriLink.getHost()))) {
+                Url.INSTANCE.getDomains().add();
+            }
+            Url.INSTANCE.getVisits().add();
+
+        } catch (IllegalArgumentException e) {
+            Log.e(LOGTAG, "Invalid URL", e);
+        }
+
+    }
+
+    public static void sessionStop() {
+        domainMap.clear();
+        Pings.INSTANCE.getSessionEnd().send();
+    }
+
+    @UiThread
+    public static void urlBarEvent(boolean aIsUrl) {
+        if (aIsUrl) {
+            Url.INSTANCE.getQueryType().get("type_link").add();
+        } else {
+            Url.INSTANCE.getQueryType().get("type_query").add();
+            // Record search engines.
+            String searchEngine = getDefaultSearchEngineIdentifierForTelemetry();
+            Searches.INSTANCE.getCounts().get(searchEngine).add();
```

Comment:
This seems to not work in Glean Debug Ping Viewer. `searchEngine` is a search engine like `www.google.com`, but in Viewer, it shows `invalid_label`.

Reply:
According to your [implementation of `getDefaultSearchEngineIdentifierForTelemetry`](https://github.com/MozillaReality/FirefoxReality/pull/2241/files#diff-cb02bef369d942cc3f925e33897d6fe7R159), you're using `getResourceURL` which returns a [full URL](https://github.com/MozillaReality/FirefoxReality/blob/7fcabcb2834bbbdf51cafb9a17a58271f6c95cbd/app/src/common/shared/org/mozilla/vrbrowser/search/SearchEngineWrapper.java#L125) for a search engine. As you can see from our docs, [labels only accept specific set of characters](https://mozilla.github.io/glean/book/user/metrics/labeled_counters.html#limits).

[Fenix](https://github.com/mozilla-mobile/fenix/blob/c6562bff98d602b969a71fba3dc84b86821251d4/app/src/main/java/org/mozilla/fenix/components/metrics/GleanMetricsService.kt#L514) uses the [lower case `identifier`](https://github.com/mozilla-mobile/fenix/blob/c6562bff98d602b969a71fba3dc84b86821251d4/app/src/main/java/org/mozilla/fenix/components/metrics/Metrics.kt#L269) as a label for counts.

You should probably do the same.